packages/blade/src/utils/useBreakpoint/useBreakpoint.ts

```diff
@@ -0,0 +1,144 @@
+import { useEffect, useState, useCallback, useMemo } from 'react';
+import type { Breakpoints } from '../../tokens/global';
+
+type DeviceType = 'desktop' | 'mobile';
+type Breakpoint = keyof Breakpoints | undefined;
+
+type BreakpointAndDevice = {
+  matchedBreakpoint: Breakpoint;
+  matchedDeviceType: DeviceType;
+};
+
+const useBreakpoint = ({ breakpoints }: { breakpoints: Breakpoints }): BreakpointAndDevice => {
+  const supportsMatchMedia =
+    typeof document !== 'undefined' && typeof window?.matchMedia === 'function';
+
+  const breakpointsTokenAndQueryCollection = useMemo(
+    () =>
+      (supportsMatchMedia
+        ? Object.entries(breakpoints).map(
+            ([breakpointTokenName, breakpointSize], index, breakpointsArray) => {
+              let mediaQuery = '';
+
+              if (breakpointTokenName === 'max') {
+                mediaQuery = `screen and (min-width: ${breakpointSize}px)`;
+              } else if (breakpointsArray[index - 1]) {
+                mediaQuery = `screen and (min-width: ${
+                  breakpointsArray[index - 1][1] + 1
+                }px) and (max-width: ${breakpointSize}px)`;
+              } else {
+                mediaQuery = `screen and (max-width: ${breakpointSize}px)`;
+              }
+
+              return [breakpointTokenName, breakpointSize, mediaQuery];
+            },
+          )
+        : []) as [keyof Breakpoints, number, string][],
+    [breakpoints, supportsMatchMedia],
+  );
```

Comment:
the output of this function will look something like following if the device supports `matchMedia`

```
[
  ['xs', '320', `screen and (max-width: 320px)`]
  ['s', '480', `screen and (min-width:321px) and (max-width: 480px`]
  .
  .
  .
  ['max', '1201', `screen and (min-width: 1201px)`]
]
```
if the matchMedia isn't supported which will be the case for `ssr` and react native then the function will return `[]`

Reply:
Write the last line in code comments.